apiv2/project/project.go

```diff
@@ -0,0 +1,540 @@
+package project
+
+import (
+	"context"
+	"errors"
+
+	modelv2 "github.com/mittwald/goharbor-client/apiv2/model"
+	uc "github.com/mittwald/goharbor-client/apiv2/user"
+
+	v2client "github.com/mittwald/goharbor-client/apiv2/internal/api/client"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client"
+
+	"github.com/go-openapi/runtime"
+	"github.com/mittwald/goharbor-client/apiv2/internal/legacyapi/client/products"
+	model "github.com/mittwald/goharbor-client/apiv2/model/legacy"
+)
+
+const (
+	ProjectMetadataKeyEnableContentTrust   MetadataKey = "enable_content_trust"
+	ProjectMetadataKeyAutoScan             MetadataKey = "auto_scan"
+	ProjectMetadataKeySeverity             MetadataKey = "severity"
+	ProjectMetadataKeyReuseSysCVEWhitelist MetadataKey = "reuse_sys_cve_whitelist"
+	ProjectMetadataKeyPublic               MetadataKey = "public"
+	ProjectMetadataKeyPreventVul           MetadataKey = "prevent_vul"
+	ProjectMetadataKeyRetentionID          MetadataKey = "retention_id"
+)
+
+// RESTClient is a subclient for handling project related actions.
+type RESTClient struct {
+	// The legacy swagger client
+	LegacyClient *client.Harbor
+
+	V2Client *v2client.Harbor
```

Comment:
Please add a godo line here and on every other subclient implementing a `V2Client`.